Skip tests that won't compile with opaque structures in libssl

Index: cachedsess.t.c
--- cachedsess.t.c.orig
+++ cachedsess.t.c
@@ -128,7 +128,7 @@ START_TEST(cache_dsess_03)
 }
 END_TEST
 
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x3040000fL)
 START_TEST(cache_dsess_04)
 {
 	SSL_SESSION *s1, *s2;
@@ -169,7 +169,7 @@ cachedsess_suite(void)
 	tcase_add_test(tc, cache_dsess_01);
 	tcase_add_test(tc, cache_dsess_02);
 	tcase_add_test(tc, cache_dsess_03);
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x3040000fL)
 	tcase_add_test(tc, cache_dsess_04);
 #endif
 	suite_add_tcase(s, tc);
